### DISCUSSION TITLE
Does NAPS2 support templates?

### DISCUSSION THREAD

**Discussion by mainrs** [cyanfish/naps2#125] [General]
My idea was to create a bunch of scripts that I can use to scan various types of documents. Alongside standard-sized papers like ISO DINX and US Letter, I would like to have more specialized templates.

One example would be to scan credit-card sized documents like driver's licenses. The user would first scan the frontside, then the backside. The sides could be scanned to an image of size 84x56mm, which is the exact size of credit-sized cards. It would be even better if this could be taken further and enable composition of scans. One could then use the two scanned images and generate a PDF file that has both of these images on a single page.

My question is: does NAPS2 support this? Even the custom size through CLI option would be enough, since the other idea can be done as a post-processing step.

**Comment by cyanfish** [comment]
Essentially yes, you can create profiles in the GUI with custom page sizes and then use those profiles from the CLI.

Combining multiple images on the same page currently isn't possible but is planned (eventually).

**Reply by mainrs** [reply]
Thank you very much!
